### PR TITLE
[DDW-254] Integrate latest cardano-js and update address validation logic

### PR DESCRIPTION
**Pull request by DeeJayElly**
This PR integrates latest cardano-js library and updates address validation logic.

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1591981536038700)

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@DeeJayElly looks like there are some build issues: 
Screenshot 2020-06-10 at 13 42 20

Can you please check those out?

File: source/renderer/app/stores/WalletsStore.js [hunk 1]
```diff
@@ -44,6 +42,7 @@ import type {
   TransferFundsCalculateFeeRequest,
   TransferFundsRequest,
 } from '../api/wallets/types';
+import { TESTNET_MAGIC } from '../../../main/config';
```
Comment (nikolaglumac) [inline source/renderer/app/stores/WalletsStore.js:45]: @DeeJayElly this breaks the build... You can NOT import anything from the `source/main` into `source/renderer`. Please move the `TESTNET_MAGIC` under `source/common` and then import it in appropriate files in main and renderer...

Comment (DeeJayElly) [reply source/renderer/app/stores/WalletsStore.js]: Fixed

**Comment by DeeJayElly** [comment]
> @DeeJayElly looks like there are some build issues:
> Screenshot 2020-06-10 at 13 42 20
> 
> Can you please check those out?

Fixed

**Comment by ManusMcCole** [comment]
I was having an issue sending funds between Rewards Wallets that both contained funds on the ITN Rewards Network. I received an invalid address error message. I ran in to some node issues before I could take a video but once resynced I will post

**Comment by ManusMcCole** [comment]
ezgif com-video-to-gif

**Comment by ManusMcCole** [comment]
Similar error on Friends and Family Testnet. No error on console 
ezgif com-video-to-gif (1)

**Comment by gnpf** [comment]
@DeeJayElly The issue @ManusMcCole reported is also found in Shelley QA build:

YBGgIJFKMn

**Comment by rhyslbw** [comment]
I suggest closing this, as https://github.com/input-output-hk/daedalus/pull/2031 supersedes

**Comment by nikolaglumac** [comment]
> I suggest closing this, as #2041 supersedes

Agreed @rhyslbw 👍

**Pull request closed without merging**